Formula/git-when-merged.rb

```diff
@@ -1,11 +1,12 @@
 class GitWhenMerged < Formula
-  include Language::Python::Shebang
+  include Language::Python::Virtualenv
 
   desc "Find where a commit was merged in git"
-  homepage "https://github.com/mhagger/git-when-merged"
-  url "https://github.com/mhagger/git-when-merged/archive/v1.2.1.tar.gz"
-  sha256 "46ba5076981862ac2ad0fa0a94b9a5401ef6b5c5b0506c6e306b76e5798e1f58"
+  homepage "https://pypi.org/project/git-when-merged/"
+  url "https://files.pythonhosted.org/packages/35/45/a2696a3a133838739baaa20fbb180b98c2c053970b2723b4384cbbc1f4a0/git-when-merged-1.2.1.tar.gz"
+  sha256 "0b1addf54e9b5e428ab577a2ace5b7ebc167d067301f96447eccfc82bbad65fe"
   license "GPL-2.0-only"
+  revision 1
```

Comment:
```suggestion
```
If there's no real difference then we don't need a `revision`.

Reply:
> In Homebrew we sometimes accept formulae updates that don’t include a version bump. These include resource updates, new patches or fixing a security issue with a formula.

doesn't upstream source (pypi) and packing content (with virtualenv) change meet the criteria of "resource updates"? I could remove this line if really unnecessary.

Reply:
Bumping the `revision` means that a formula gets reinstalled by `brew upgrade`. Since this change should have no user-facing changes (i.e. functionality, etc, are all exactly the same as before your PR), that's not necessary.

Reply:
fine, but I am still confused about the usage of `revision`, as a user I read revision number like "there is tweaks on the homebrew side, no upstream change".

Reply:
This might be the case, but this might also be because dependencies were updated and that the formula can't work anymore if not recompiled with the new dependencies (which is why sometimes formula update revision bumps other formulas).
You can also see this on Linux distros.

Reply:
The correct way to read that is to slightly amend it: "there are user-facing tweaks on the homebrew side, no upstream change".

If we update the formula's `test`, for example, then there is no need to bump the `revision`, since that will have zero impact on anything a user has installed.

Reply:
I get it, fixed.

Reply:
If this is the homebrew community approach to versioning, I could just follow.
Though this still doesn't align with my understanding of package versioning, the version number should capture a unique state of the package, and I would be surprised if the package content changed under an identical version number.

Reply:
But the content doesn't change here. Just the source of the content

Reply:
depends on what the definition of "package content" is:
- the application function being shipped: no change, user faced functionality is the same.
- the `git-when-merged` binary: changed from whole script to a setuptools plus virtualenv generated one, sth like `from git_when_merged import main; sys.exit(main())`.
- bottle: changed, of course.

Reply:
@carlocab is there anything else to have this merged?

Reply:
Ah, I thought I had explained this already: we don't want to use `virtualenv_install_with_resources` here.

Reply:
without `virtualenv_install_with_resources`, how to write a formula to install a python package, I don't understand... and didn't find that review comment.

Reply:
Sorry, that was what I meant [here](https://github.com/Homebrew/homebrew-core/pull/107876#discussion_r944363067), as `virtualenv_install_with_resources` uses `pip` under the hood.

You can leave the install method as it was before your PR. The PyPI tarball is the same as the GitHub one -- the only difference is some comments.

Reply:
> You can leave the install method as it was before your PR. The PyPI tarball is the same as the GitHub one -- the only difference is some comments.

why not install it as a proper python package but symlink to a script, I consider it a kind of programming by coincidence.
my reasoning: this is a python application published as a python package on pypi, to install a python package on pypi, you use pip.

Reply:
And my reasoning is that doing that has negligible desirable consequences but significant undesirable ones.

This is why I asked if it looked like they would only support `pip` installs (or some other PEPxyz-compliant mechanism) soon. If that were the case, then the argument in favour of using `pip` is a whole lot more compelling. But it doesn't seem like it is, so it's not very compelling.

Note that this is hardly programming-by-coincidence: our choice of installation method is explicitly supported upstream. See options 3 and 4 in the [installation instructions of the README](https://github.com/mhagger/git-when-merged#installation).

Reply:
ok, I will close this PR until upstream changes (very unlikely).

Reply:
Thanks for the PR anyway, @georgexsh!